azurerm/resource_arm_sql_database.go

```diff
@@ -118,128 +119,157 @@ func resourceArmSqlDatabase() *schema.Resource {
 	}
 }
 
-func resourceArmSqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmSqlDatabaseCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	sqlDatabasesClient := meta.(*ArmClient).sqlDatabasesClient
 
-	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	command := &sql.CreateOrUpdateDatabase{
-		Name:              d.Get("name").(string),
-		Location:          d.Get("location").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ServerName:        d.Get("server_name").(string),
-		Tags:              *expandedTags,
-		CreateMode:        azure.String(d.Get("create_mode").(string)),
-	}
+	databaseName := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	serverName := d.Get("server_name").(string)
+	sourceDatabaseID := d.Get("source_database_id").(string)
+	collation := d.Get("collation").(string)
+	elasticPoolName := d.Get("elastic_pool_name").(string)
 
-	if v, ok := d.GetOk("source_database_id"); ok {
-		command.SourceDatabaseID = azure.String(v.(string))
-	}
+	createMode := sql.CreateMode(d.Get("create_mode").(string))
 
-	if v, ok := d.GetOk("edition"); ok {
-		command.Edition = azure.String(v.(string))
-	}
+	maxSizeBytes := d.Get("max_size_bytes").(string)
 
-	if v, ok := d.GetOk("collation"); ok {
-		command.Collation = azure.String(v.(string))
-	}
+	edition := sql.DatabaseEdition(d.Get("edition").(string))
```

Comment:
given this field is optional, I think we want to optionally assign it to `props` below? i.e.

```
if v, ok := d.GetOk("edition"); ok {
  props.Edition = sql.DatabaseEdition(v.(string))
}
```